equinox/nn/pool.py

```diff
@@ -256,3 +256,35 @@ def __init__(
             padding=padding,
             **kwargs,
         )
+
+
+class AdaptiveAvgPool1d(Module):
+    """Adaptive 1D downsampling for a target shape."""
+
+    target_size: int = static_field()
+
+    def __init__(self, target_size: int):
+        """**Arguments:**
+        - `target_size`: The target output size.
+        """
+
+        self.target_size = target_size
+
+    def __call__(self, x: Array):
+        assert (
+            x.ndim == 1
+        ), f"Only supports 1D input, received input with {x.ndim} dimensions."
+        channels = jnp.size(x)
+        assert (
+            channels >= self.target_size
+        ), f"Final Pooled size {self.target_size} cannot be greater than input size {channels}."
+
+        splits = jnp.array_split(x, self.target_size)
+        num_head_arrays = channels % self.target_size
+        if num_head_arrays:
+            head_mean = jax.vmap(jnp.mean)(jnp.asarray(splits[:num_head_arrays]))
+            tail_mean = jax.vmap(jnp.mean)(jnp.asarray(splits[num_head_arrays:]))
```

Comment:
This is going to be quite inefficient: `x` is split into several arrays (which can be done as views and is efficient), and then concatenated back together into two groups (in the `jnp.asarray`; and this isn't efficient).

I think it should be possible to first split `x` into just the head and tail piece (which can be done as views and is efficient), and then reshape each piece so that they have a leading dimension of the appropriate size (again, this is efficient), and then vmap'ing the mean operation over it.

In each case above, "efficient" = "not copying the underlying data".

Reply:
I am not sure if I understood it properly. At the moment, splits is a list of arrays. So I understood that you meant to get rid of it altogether and apply the vmap on the 2 parts of the input array. However, can you elaborate on how to use `view` to do this ? Alternatively, I have tried it with `reshape` and it is definitely faster than the older implementation.

Meanwhile, I tested the older implementation with the new one 
```
`new`: 5 loops, best of 5: 1.98 ms per loop
`old`: 5 loops, best of 5: 2.53 s per loop
```